sql-to-dbsp-compiler/lib/sqllib/src/casts.rs

```diff
@@ -222,52 +222,68 @@ pub fn cast_to_decimal_bN(value: Option<bool>, precision: u32, scale: i32) -> De
 }
 
 #[inline]
-pub fn cast_to_decimal_decimal(value: Decimal, _precision: u32, scale: i32) -> Decimal {
-    //value.with_prec(precision as u64).with_scale(scale as i64)
+pub fn cast_to_decimal_decimal(value: Decimal, precision: u32, scale: u32) -> Decimal {
+    // make sure we can fit the left half of the number in the new wanted precision
+
+    // '1234.5678' -> DECIMAL(6, 2) is fine as the integer part fits in 4 digits
+    // but to DECIMAL(6, 3) would error as we can't fit '1234' in 3 digits
+
+    let int_part_precision = value
+        .trunc()
+        .mantissa()
+        .checked_ilog10()
+        .map(|v| v + 1)
+        .unwrap_or(0);
+    let to_int_part_precision = precision - scale;
+
+    assert!(
```

Comment:
Assert is just for debugging. If an assertion fails it should indicate a bug.
Assertion failures should never be caused by user input. 
Here you should probably panic for now, until we implement the error return API.